pkg/cluster/manager/edit_config.go

```diff
@@ -69,34 +74,40 @@ func (m *Manager) EditConfig(name string, skipConfirm bool) error {
 	return nil
 }
 
+// If the flag --topology-file is specified, the first 2 steps will be skipped.
 // 1. Write Topology to a temporary file.
 // 2. Open file in editor.
 // 3. Check and update Topology.
 // 4. Save meta file.
-func (m *Manager) editTopo(origTopo spec.Topology, data []byte, skipConfirm bool) (spec.Topology, error) {
-	file, err := os.CreateTemp(os.TempDir(), "*")
-	if err != nil {
-		return nil, perrs.AddStack(err)
-	}
+func (m *Manager) editTopo(origTopo spec.Topology, data []byte, opt EditConfigOptions, skipConfirm bool) (spec.Topology, error) {
+	var name string
+	if len(opt.NewTopoFile) == 0 {
```

Comment:
I think it's safe to check `if opt.NewTopoFile == ""`

Reply:
done